### PR TITLE
Add some doc aliases to Vec

### PR DESCRIPTION
**Pull request by sugar700**
Fixes https://twitter.com/artichokeruby/status/1312505176167645189

### CODE REVIEW

**Comment by xfix** [comment]
Actually, may as well add `ArrayList` and `List` too. `ArrayList` comes from Java, and `List` comes from Python.

File: library/alloc/src/vec.rs [hunk 1]
```diff
@@ -296,6 +296,9 @@ use crate::raw_vec::RawVec;
 /// [owned slice]: Box
 /// [slice]: ../../std/primitive.slice.html
 /// [`&`]: ../../std/primitive.reference.html
+#[doc(alias = "ArrayList")]
+#[doc(alias = "List")]
+#[doc(alias = "Vector")]
```
Comment (pickfire) [inline library/alloc/src/vec.rs:301]:
```suggestion
#[doc(alias = "array_list")]
#[doc(alias = "list")]
#[doc(alias = "vector")]
```
We should keep it `snake_case` like rust style, python also called it `list`, not `List` (although that is the typing type).

Comment (pickfire) [reply library/alloc/src/vec.rs:301]: Oh, I forgot this is a struct, can just ignore it. Sorry.

**Comment by xfix** [comment]
Weird, there apparently was merge conflict, but using `git pull --rebase origin master` automatically dealt with it.

**Comment by camelid** [comment]
cc #81989

**Comment by JohnCSimon** [comment]
@xfix  - Ping from triage, can you please resolve the merge conflict? Thank you. 
@rustbot label: -S-waiting-on-review +S-waiting-on-author

**Comment by camelid** [comment]
@xfix Please [rebase](https://rustc-dev-guide.rust-lang.org/git.html#rebasing) instead of merging; rust-lang/rust follows a "no-merge policy". Thank you!

**Comment by m-ou-se** [comment]
Thanks for your PR and sorry for the late reply. This has been sitting still without review because it's not quite clear what exactly we want to use `doc(alias)` for.

See https://github.com/rust-lang/rust/pull/81989#issuecomment-824143364.

While I do personally thik that `doc(alias = "vector")` would be a good one to add (since that's how we pronounce 'Vec' most of the time), we'd first like to reach some agreement on what kind of aliases we want and don't want before continuing merging PRs like this, as mentioned in the linked comment:

> [..] we want some kind of clear guideline/policy that we can all agree on before accepting more aliases. Feel free to open an issue on https://github.com/rust-lang/libs-team or ping us on Zulip if you (anyone) want to work on this.

So, I'm closing this PR for now. Once we have a policy in place and this PR fits within it, feel free to re-open it.

**Pull request closed without merging**